source/includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/v2/bson"
+	"go.mongodb.org/mongo-driver/v2/mongo"
+	"go.mongodb.org/mongo-driver/v2/mongo/options"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Replace the placeholder with your Atlas connection string
+	const uri = "mongodb+srv://user:123@atlascluster.spm1ztf.mongodb.net/?retryWrites=true&w=majority&appName=AtlasCluster"
+
+	// Connect to your Atlas cluster
+	clientOptions := options.Client().ApplyURI(uri)
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		log.Fatalf("failed to connect to the server: %v", err)
+	}
+	defer func() { _ = client.Disconnect(ctx) }()
+
+	// Set the namespace
+	coll := client.Database("sample_mflix").Collection("embedded_movies")
+
+	// start-create-vector-search
+	// Defines the structs used for the index definition
+	type vectorDefinitionField struct {
+		Type          string `bson:"type"`
+		Path          string `bson:"path"`
+		NumDimensions int    `bson:"numDimensions"`
+		Similarity    string `bson:"similarity"`
+		Quantization  string `bson:"quantization"`
+	}
+
+	type vectorDefinition struct {
+		Fields []vectorDefinitionField `bson:"fields"`
+	}
+
+	// Sets the index name and type to "vectorSearch"
+	indexName := "vector_index"
+	opts := options.SearchIndexes().SetName(indexName).SetType("vectorSearch")
+
+	// Defines the index definition
+	vectorSearchIndexModel := mongo.SearchIndexModel{
+		Definition: vectorDefinition{
+			Fields: []vectorDefinitionField{{
+				Type:          "vector",
+				Path:          "plot_embedding",
+				NumDimensions: 1536,
+				Similarity:    "dotProduct",
+				Quantization:  "scalar"}},
+		},
+		Options: opts,
+	}
+
+	// Creates the index
+	searchIndexName, err := coll.SearchIndexes().CreateOne(ctx, vectorSearchIndexModel)
+	if err != nil {
+		log.Fatalf("failed to create the vector search index: %v", err)
+	}
+	// end-create-vector-search
+
+	// Creates an Atlas Search index
+	// start-create-atlas-search
+	// Sets the index name and type to "search"
+	indexName := "atlas_search_index"
+	opts := options.SearchIndexes().SetName(indexName).SetType("search")
+
+	// Defines the index definition 
+	searchIndexModel := mongo.SearchIndexModel{
+		Definition: bson.D{
+			{Key: "mappings", Value: bson.D{
+				{Key: "dynamic", Value: false},
+				{Key: "fields", Value: bson.D{
+					{Key: "plot", Value: bson.D{
+						{Key: "type", Value: "string"},
+					}},
+				}},
+			}},
+		},
+		Options: opts,
+	}
+
+	// Creates the index
+	searchIndexName, err := coll.SearchIndexes().CreateOne(ctx, searchIndexModel)
+	if err != nil {
+		log.Fatalf("failed to create the atlas search index: %v", err)
+	}
+	// end-create-atlas-search
+
+	// start-list-index
+	// Specify the index to retrieve
```

Comment:
```suggestion
	// Specifies the index to retrieve
```
Nit: keeps it in the same tense as rest of comments

Reply:
See above comment